### PR TITLE
initial refactor of fileHandler to ts

### PR DESCRIPTION
**Pull request by terakilobyte**
Pulled hard on a string

### CODE REVIEW

File: src/codeGenerator.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ export interface CodeFile {
 
 export function buildCodeFiles(source: string, type: string): CodeFile {
   const sourceLines = source.split("\n");
-  let fileType: string = null;
+  const fileType = type.substr(1);
```
Comment (cbush) [inline src/codeGenerator.ts:11]: Why this change? This breaks the buildCodeFiles tests

Comment (cbush) [inline src/codeGenerator.ts:11]:
```suggestion
  const fileType = type;
```

File: src/codeGenerator.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ export interface CodeFile {
 
 export function buildCodeFiles(source: string, type: string): CodeFile {
   const sourceLines = source.split("\n");
-  let fileType: string = null;
```
Comment (cbush) [inline src/codeGenerator.ts:11]: Missed this on the last one.

**Review by cbush — APPROVED** [review]
Looks good pending the acceptance of the suggestion (tests work with that change)

File: src/codeGenerator.ts (hunk 1 above)
Comment (terakilobyte) [reply src/codeGenerator.ts]: jest autorunner failed silently :angry: 

I made this change erroneously seeing another error somewhere else. Fixed
